src/main/scala/com/redislabs/provider/redis/rdd/RedisRDD.scala

```diff
@@ -98,6 +98,33 @@ class RedisListRDD(prev: RDD[String],
   }
 }
 
+class RedisPFCountRDD(prev: RDD[String])(implicit val readWriteConfig: ReadWriteConfig)
+  extends RDD[(String,Long)](prev) with Keys {
+  override def getPartitions: Array[Partition] = prev.partitions
+  override def compute(split: Partition, context: TaskContext): Iterator[(String,Long)] = {
+    val partition: RedisPartition = split.asInstanceOf[RedisPartition]
+    val sPos = partition.slots._1
+    val ePos = partition.slots._2
+    val nodes = partition.redisConfig.getNodesBySlots(sPos, ePos)
+    val keys = firstParent[String].iterator(split, context)
+    getPFCOUNT(nodes,keys)
+  }
+  def getPFCOUNT(nodes: Array[RedisNode], keys: Iterator[String]): Iterator[(String,Long)] = {
+    groupKeysByNode(nodes, keys).flatMap { x => {
+        val conn = x._1.endpoint.connect()
+        val stringKeys = filterKeysByType(conn, x._2, "string")
+        val pipeline = conn.pipelined
```

Comment:
Please refactor to use `mapWithPipeline` utility function (was introduced after the origin PR):

```
 val response = mapWithPipeline(conn, stringKeys) { (pipeline, key) =>
    // scala gives compilation error for string and varargs, if we use pfcount(<string>)
    pipeline.pfcount(Seq(key): _*)
}
val res = stringKeys.zip(response).iterator.filter { _._2.isInstanceOf[Long] }
          .asInstanceOf[Iterator[(String,Long)]]
```